src/aof.c

```diff
@@ -1549,18 +1548,19 @@ int loadAppendOnlyFiles(aofManifest *am) {
         serverAssert(am->base_aof_info->file_type == AOF_FILE_TYPE_BASE);
         aof_name = (char*)am->base_aof_info->file_name;
         updateLoadingFileName(aof_name);
-        ++aof_num;
+        last_file = ++aof_num == total_num;
         start = ustime();
         ret = loadSingleAppendOnlyFile(aof_name);
-        if (ret == AOF_OK || (ret == AOF_TRUNCATED && aof_num == total_num)) {
+        if (ret == AOF_OK || (ret == AOF_TRUNCATED && last_file)) {
             serverLog(LL_NOTICE, "DB loaded from base file %s: %.3f seconds",
                 aof_name, (float)(ustime()-start)/1000000);
-            ret = AOF_OK;
         }
 
         /* If an AOF exists in the manifest but not on the disk, Or the truncated 
          * file is not the last file, we consider this to be a fatal error. */
-        if (ret == AOF_NOT_EXIST || ret == AOF_TRUNCATED) ret = AOF_FAILED;
+        if (ret == AOF_NOT_EXIST || (ret == AOF_TRUNCATED && !last_file)) {
+            ret = AOF_FAILED;
+        }
 
         if (ret != AOF_OK && ret != AOF_EMPTY) {
```

Comment:
need to add TRUNCATED here, or maybe just change it to an `else` for the above?

Reply:
Changed to if (ret == AOF_OPEN_ERR || ret == AOF_FAILED)

Reply:
is any of the other states valid in this case? like NOT_EXIST and OPEN_ERR?

Reply:
#define AOF_OK 0     -> OK
#define AOF_NOT_EXIST 1  -> AOF_FAILED -> goto cleanup
#define AOF_EMPTY 2  -> OK
#define AOF_OPEN_ERR 3 -> goto cleanup
#define AOF_FAILED 4 -> goto cleanup
#define AOF_TRUNCATED 5 -> AOF_FAILED (if not last file) -> goto cleanup

Reply:
I think I covered all the status

Reply:
ok, i think you're right. wanted to double check.